### PR TITLE
Support multiple indexer-service endpoints

### LINKED ISSUE #48: Combine query count from multiple indexer-service instances
Current implementation supports only a single indexer service endpoint, this is not ideal for setups that scale horizontally with multiple indexer-service instances. While most of the time the traffic is shared evenly between the instances, allowing the learning process to get the feedback it requires, this is not always optimal.

- Inaccurate statistics: If the load is not evenly distributed among all instances, the query count from a single instance may not be representative of the overall query count. Think hash-based algorithm , rules based on specific URL, or primary/secondary setups
- Lack of granularity of low traffic subgraphs

### PR DESCRIPTION
**Pull request by madumas**
To fix issue #48, this allows specify more than one endpoint in `--indexer-service-metrics-endpoint`. `subgraph_query_count` sums the metrics from the different endpoints for a given subgraph.

Also removes some extra code and add tests.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -36,6 +36,7 @@ prometheus-async = {extras = ["aiohttp"], version = "^22.1.0"}
 backoff = "^2.2.1"
 python-json-logger = "^2.0.4"
 autoagora-agents = {git = "https://github.com/semiotic-ai/autoagora-agents.git", tag = "v1.0.1"}
+vcrpy = "^4.2.1"
```
Comment (aasseman) [inline pyproject.toml:39]: `vcrpy` should be moved into the  dev dependencies below

**Review by aasseman — CHANGES_REQUESTED** [review]
Very nice! Thanks!
Could you also run black / isort / pyright?

**Review by aasseman — CHANGES_REQUESTED** [review]
There is an issue with the history, probably due to the merging of master into this branch. That caused some things to get un-removed (such as `vcrpy` in the main dependencies, or line 114 in `config.py`).
I would suggest an interactive rebase onto master to fix the history.

In general, catching up to the target branch through rebasing is safer and cleaner than merging.

**Comment by madumas** [comment]
it should be fine now

**Review by aasseman — APPROVED** [review]
Thanks for bearing with me through the review process! Your contribution is much appreciated.
Also I really like the tests using `vcrpy` :slightly_smiling_face: .

**Pull request merged**